soc/arm/microchip_mec/mec1501/power.c

```diff
@@ -101,9 +101,9 @@ static void z_power_soc_sleep(void)
  * For deep sleep pm_system_suspend has executed all the driver
  * power management call backs.
  */
-__weak void pm_power_state_set(struct pm_state_info info)
+__weak void pm_power_state_set(struct pm_state_info *info)
```

Comment:
Some API calls like this one use, in practice, the `state` field of `struct pm_state_info` (not sure why `struct pm_state_info` is really here). Since this is changing the API signature, I think it's worth investigating if we can have instead:

```c
void pm_power_state_set(enum pm_state state)
```

Reply:
We may also pass that one (a state is in reality made by state + substate...). Just saying because the info structure contains other unrelated information (like residency times) that are never used by these calls. It makes the API confusing IMO.

Reply:
done